<div class="alert alert-block alert-info">
<center>ESTATÍSTICA APLICADA</center>
<center>PROJETO 1</center>
    <center>PROFESSOR: Paulo Ribeiro Lins Junior</center>
<center>ALUNO: Matheus Carneiro da Cunha</center>
</div>

In [1]:
import pandas as pd

# Tarefa 1 Carregando os DataFrames

### Sim é possivel carregar dataframes sem precisar baixar os arquivos, estou puxando-os direto do meu repositorio no Github, tambem usamos alguns parametros para evitar erros de visualização, sendo eles o encoding = latan1, pois os dados são de locais na america latina, e o de separador (sep) = ';' (ponto e virgula).

In [2]:
df18 = pd.read_csv("https://raw.githubusercontent.com/MatheusCarne/projeto1fpr/main/datatran2018.csv",encoding='latin1', sep=';')
df19 = pd.read_csv("https://raw.githubusercontent.com/MatheusCarne/projeto1fpr/main/datatran2019.csv",encoding='latin1',  sep=';')
df20 = pd.read_csv("https://raw.githubusercontent.com/MatheusCarne/projeto1fpr/main/datatran2020.csv",encoding='latin1',  sep=';')

In [3]:
df18

id data_inversa     dia_semana   horario  uf     br     km  \
0       99973.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
1       99976.0   2018-01-01  segunda-feira  00:40:00  SC  282.0    0,4   
2       99977.0   2018-01-01  segunda-feira  00:30:00  RJ  493.0      1   
3       99981.0   2018-01-01  segunda-feira  01:15:00  RS  386.0    134   
4       99982.0   2018-01-01  segunda-feira  00:20:00  RS  293.0  151,7   
...         ...          ...            ...       ...  ..    ...    ...   
69290  252256.0   2018-12-24  segunda-feira  09:15:00  PI  343.0  263,1   
69291  252652.0   2018-12-12   quarta-feira  07:00:00  BA  101.0    321   
69292  252801.0   2018-11-15   quinta-feira  10:48:00  SP  116.0    529   
69293  253040.0   2018-11-23    sexta-feira  19:30:00  RS  472.0  573,5   
69294  253078.0   2018-10-01  segunda-feira  04:31:00  GO   40.0      9   

                       municipio  \
0                        RESENDE   
1                  FLORIANOPOLIS   
2                       ITABORAI   
3                        SARANDI   
4                       CANDIOTA   
...                          ...   
69290                CAMPO MAIOR   
69291  PRESIDENTE TANCREDO NEVES   
69292             BARRA DO TURVO   
69293                 URUGUAIANA   
69294                   LUZIANIA   

                                          causa_acidente  \
0                                      Condutor Dormindo   
1                     Não guardar distância de segurança   
2                                 Ultrapassagem Indevida   
3                                     Ingestão de Álcool   
4                            Falta de Atenção à Condução   
...                                                  ...   
69290                                   Agressão Externa   
69291                        Falta de Atenção à Condução   
69292                        Falta de Atenção à Condução   
69293  Desobediência às normas de trânsito pelo condutor   
69294                 Não guardar distância de segurança   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0         Saída de leito carroçável  ...              4      1         0   
1                  Colisão traseira  ...              2      1         0   
2                   Colisão frontal  ...              0      3         1   
3               Colisão transversal  ...              0      2         0   
4         Saída de leito carroçável  ...              1      0         0   
...                             ...  ...            ...    ...       ...   
69290   Colisão com objeto estático  ...              0      1         0   
69291     Saída de leito carroçável  ...              0      0         0   
69292              Colisão traseira  ...              0      0         1   
69293  Queda de ocupante de veículo  ...              1      1         0   
69294     Saída de leito carroçável  ...              0      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           4        1     -22,46937     -44,44705     SR-RJ     DEL5/7   
1           2        2  -27,59971687  -48,57565694     SR-SC     DEL8/1   
2           1        3    -22,763901    -42,927532     SR-RJ     DEL5/4   
3           0        2    -27,953636    -52,916374     SR-RS    DEL9/14   
4           1        1    -31,395214  -53,78391174     SR-RS    DEL9/11   
...       ...      ...           ...           ...       ...        ...   
69290       0        1     -4,819326    -42,158821     SR-PI    DEL17/1   
69291       1        1  -13,44768518  -39,42066193     SR-BA    DEL10/5   
69292       1        2    -24,939211    -48,301248     SR-SP     DEL6/5   
69293       1        2  -29,76355779  -57,05200195     SR-RS    DEL9/13   
69294       1        1    -16,131158    -47,957739     SR-DF   UOP01/DF   

            uop  
0      UOP03/RJ  
1      UOP01/SC  
2      UOP02/RJ  
3      UOP01/RS  
4      UOP03/RS  
...         ...  
69290  UOP03/PI  


In [4]:
df19

id data_inversa     dia_semana   horario  uf     br     km  \
0      182210.0   2019-01-01    terça-feira  01:30:00  SP  116.0    218   
1      182211.0   2019-01-01    terça-feira  01:30:00  PR  373.0  177,3   
2      182212.0   2019-01-01    terça-feira  00:45:00  SC  101.0     16   
3      182214.0   2019-01-01    terça-feira  00:00:00  CE   20.0  429,5   
4      182215.0   2019-01-01    terça-feira  01:00:00  MG  381.0    327   
...         ...          ...            ...       ...  ..    ...    ...   
67441  266255.0   2019-10-07  segunda-feira  09:10:00  ES  101.0  270,5   
67442  266406.0   2019-11-25  segunda-feira  07:20:00  PR  116.0    125   
67443  266434.0   2019-06-03  segunda-feira  19:00:00  CE  222.0    1,4   
67444  266573.0   2019-07-13         sábado  19:35:00  PR  373.0    425   
67445  266627.0   2019-10-22    terça-feira  20:20:00  BA  324.0    608   

                municipio                       causa_acidente  \
0               GUARULHOS          Falta de Atenção à Condução   
1            PONTA GROSSA          Falta de Atenção à Condução   
2                  GARUVA                     Animais na Pista   
3               FORTALEZA  Ingestão de Substâncias Psicoativas   
4                NOVA ERA          Falta de Atenção à Condução   
...                   ...                                  ...   
67441               SERRA          Falta de Atenção à Condução   
67442  FAZENDA RIO GRANDE          Falta de Atenção à Condução   
67443             CAUCAIA   Não guardar distância de segurança   
67444              CANDOI          Defeito Mecânico no Veículo   
67445        SIMOES FILHO                       Defeito na Via   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0       Colisão com objeto estático  ...              0      0         0   
1                  Colisão traseira  ...              0      0         0   
2       Colisão com objeto estático  ...              0      0         0   
3       Colisão com objeto estático  ...              1      0         0   
4               Colisão transversal  ...              1      1         0   
...                             ...  ...            ...    ...       ...   
67441                    Tombamento  ...              1      1         0   
67442               Colisão lateral  ...              0      1         1   
67443              Colisão traseira  ...              1      1         0   
67444     Saída de leito carroçável  ...              1      0         0   
67445  Queda de ocupante de veículo  ...              0      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           4        1  -23,46052014  -46,48772478     SR-SP     DEL6/1   
1           1        1  -25,05533957  -50,22776753     SR-PR     DEL7/3   
2           1        1      -26,1216      -48,8826     SR-SC     DEL8/3   
3           1        1      -3,76999    -38,670063     SR-CE    DEL16/1   
4           1        2      -19,7609      -43,0306     SR-MG     DEL4/3   
...       ...      ...           ...           ...       ...        ...   
67441       1        2     -20,23107     -40,27387     SR-ES    DEL12/2   
67442       1        2     -25,59505     -49,31631     SR-PR     DEL7/1   
67443       1        2     -3,736507     -38,65337     SR-CE    DEL16/1   
67444       1        1      -25,7118     -52,18374     SR-PR     DEL7/3   
67445       1        1     -12,78156     -38,41656     SR-BA    DEL10/1   

            uop  
0      UOP01/SP  
1      UOP01/PR  
2      UOP01/SC  
3      UOP02/CE  
4      UOP01/MG  
...         ...  
67441  UOP01/ES  
67442  UOP03/PR  
67443  UOP01/CE  
67444  UOP02/PR  
67445  UOP01/BA  

[67446 rows x 30 columns]

In [5]:
df20

id data_inversa    dia_semana   horario  uf     br     km  \
0      260031.0   2020-01-01  quarta-feira  01:00:00  TO  153.0  678,1   
1      260036.0   2020-01-01  quarta-feira  01:00:00  RJ  116.0    178   
2      260037.0   2020-01-01  quarta-feira  01:52:00  SC  101.0  206,9   
3      260038.0   2020-01-01  quarta-feira  01:15:00  RO  364.0    236   
4      260039.0   2020-01-01  quarta-feira  01:30:00  PR  153.0    360   
...         ...          ...           ...       ...  ..    ...    ...   
63543  364615.0   2020-03-20   sexta-feira  09:00:00  RN  304.0  171,8   
63544  364810.0   2020-09-12        sábado  23:50:00  BA  116.0   11,5   
63545  364817.0   2020-10-30   sexta-feira  10:30:00  BA  116.0   14,3   
63546  364835.0   2020-09-20       domingo  23:00:00  BA  116.0      8   
63547  364903.0   2020-10-31        sábado  02:10:00  BA  116.0      3   

                  municipio                causa_acidente  \
0                    GURUPI              Animais na Pista   
1               NOVA IGUACU  Falta de Atenção do Pedestre   
2                  SAO JOSE            Ingestão de Álcool   
3                    CACOAL       Velocidade Incompatível   
4                  REBOUCAS            Ingestão de Álcool   
...                     ...                           ...   
63543               ANGICOS   Falta de Atenção à Condução   
63544  VITORIA DA CONQUISTA            Ingestão de Álcool   
63545  VITORIA DA CONQUISTA   Falta de Atenção à Condução   
63546  VITORIA DA CONQUISTA       Velocidade Incompatível   
63547  VITORIA DA CONQUISTA              Animais na Pista   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0           Atropelamento de Animal  ...              5      0         0   
1         Atropelamento de Pedestre  ...              0      1         0   
2                   Colisão lateral  ...              0      1         0   
3         Saída de leito carroçável  ...              1      0         0   
4                       Capotamento  ...              4      0         0   
...                             ...  ...            ...    ...       ...   
63543              Colisão traseira  ...              1      2         0   
63544               Colisão lateral  ...              0      2         0   
63545               Colisão lateral  ...              0      2         0   
63546  Queda de ocupante de veículo  ...              0      0         0   
63547       Atropelamento de Animal  ...              1      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           5        1  -11,77460203  -49,10744996     SR-TO   UOP01/TO   
1           0        1  -22,75223028   -43,4379103     SR-RJ     DEL5/1   
2           1        2  -27,59193546  -48,61824557     SR-SC     DEL8/1   
3           1        1  -11,44624577  -61,43761218     SR-RO    DEL21/2   
4           6        1  -25,67503796  -50,75089805     SR-PR     DEL7/3   
...       ...      ...           ...           ...       ...        ...   
63543       2        4   -5,68711053    -36,419458     SR-RN      UOP03   
63544       1        2  -14,87308214  -40,87141514     SR-BA      UOP01   
63545       2        2  -14,89597653  -40,86098671     SR-BA      UOP01   
63546       1        1  -14,84732305  -40,88634968     SR-BA      UOP01   
63547       1        1    -14,824848    -40,861311     SR-BA      UOP01   

            uop  
0           NaN  
1      UOP02/RJ  
2      UOP01/SC  
3      UOP02/RO  
4      UOP04/PR  
...         ...  
63543     DEL01  
63544     DEL08  
63545     DEL08  
63546     DEL08  
63547     DEL08  

[63548 rows x 30 columns]

# Tarefa 2 Explorando os DataFrames

### Para poder responder as alternativas a) e b) da tarefa 2 usarei a função info do pandas nos 3 DataFrames, df18, df19 e df20

# a) quantos acidentes ocorreram em cada ano?

In [6]:
df18.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 69295 entries, 0 to 69294
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      69295 non-null  float64
 1   data_inversa            69295 non-null  object 
 2   dia_semana              69295 non-null  object 
 3   horario                 69295 non-null  object 
 4   uf                      69295 non-null  object 
 5   br                      69181 non-null  float64
 6   km                      69181 non-null  object 
 7   municipio               69295 non-null  object 
 8   causa_acidente          69295 non-null  object 
 9   tipo_acidente           69295 non-null  object 
 10  classificacao_acidente  69295 non-null  object 
 11  fase_dia                69295 non-null  object 
 12  sentido_via             69295 non-null  object 
 13  condicao_metereologica  69295 non-null  object 
 14  tipo_pista              69295 non-null

### Podemos ver que em 2018 tivemos 69295 acidentes contabiizados na tabela onde vai de 0 ao 69294.

In [7]:
df19.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67446 entries, 0 to 67445
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      67446 non-null  float64
 1   data_inversa            67446 non-null  object 
 2   dia_semana              67446 non-null  object 
 3   horario                 67446 non-null  object 
 4   uf                      67446 non-null  object 
 5   br                      67351 non-null  float64
 6   km                      67351 non-null  object 
 7   municipio               67446 non-null  object 
 8   causa_acidente          67446 non-null  object 
 9   tipo_acidente           67446 non-null  object 
 10  classificacao_acidente  67446 non-null  object 
 11  fase_dia                67446 non-null  object 
 12  sentido_via             67446 non-null  object 
 13  condicao_metereologica  67446 non-null  object 
 14  tipo_pista              67446 non-null

### Em 2019 tivemos 67446 acidentes registrados.

In [8]:
df20.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 63548 entries, 0 to 63547
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      63548 non-null  float64
 1   data_inversa            63548 non-null  object 
 2   dia_semana              63548 non-null  object 
 3   horario                 63548 non-null  object 
 4   uf                      63548 non-null  object 
 5   br                      63390 non-null  float64
 6   km                      63390 non-null  object 
 7   municipio               63548 non-null  object 
 8   causa_acidente          63548 non-null  object 
 9   tipo_acidente           63548 non-null  object 
 10  classificacao_acidente  63548 non-null  object 
 11  fase_dia                63548 non-null  object 
 12  sentido_via             63548 non-null  object 
 13  condicao_metereologica  63548 non-null  object 
 14  tipo_pista              63548 non-null

### E por fim o ano de 2020 que contabilizou 63548 acidentes no ano.

# b) quantas variáveis cada conjunto de dados registra?

### Podemos ver tambem com o comando .info() que os 3 Dataframes contem os mesmos numero de colunas, ou seja, de variaveis, que são 30 ao total.

# c) quais as cinco cidades brasileiras onde mais ocorreram acidentes em rodovias federais?

In [9]:
maisacidentes=df18['municipio'].value_counts()
df1=pd.DataFrame(maisacidentes).rename(columns={'municipio': 'acidentes'})
df1.index.names = ['Municipios']
df1.head(5)

acidentes
Municipios           
CURITIBA         1019
SAO JOSE          874
BRASILIA          867
GUARULHOS         755
PALHOCA           688

In [10]:
maisacidentes2=df19['municipio'].value_counts()
df1_1=pd.DataFrame(maisacidentes2).rename(columns={'municipio': 'acidentes'})
df1_1.index.names = ['Municipios']
df1_1.head(5)

acidentes
Municipios           
CURITIBA         1094
BRASILIA         1090
SAO JOSE          819
GUARULHOS         717
SERRA             633

In [11]:
maisacidentes3=df20['municipio'].value_counts()
df1_2=pd.DataFrame(maisacidentes3).rename(columns={'municipio': 'acidentes'})
df1_2.index.names = ['Municipios']
df1_2.head(5)

acidentes
Municipios           
BRASILIA         1040
CURITIBA          950
GUARULHOS         639
SAO JOSE          629
PALHOCA           593

# d) quantos acidentes com feridos graves aconteceram na Paraíba em 2019?

In [12]:
df2= pd.DataFrame(df19[['uf', 'feridos_graves']])
df_pb = df2[df2.uf.isin(['PB'])]
df_feridos=df_pb['feridos_graves'] > 0
qtd_feridos_pb=df_pb[df_feridos]
qtd_acidentes = qtd_feridos_pb['uf'].value_counts()
qtd_acidentes

PB    414
Name: uf, dtype: int64

### Houve 414 acidentes com feridos graves no estado da Paraíba em 2019.

# Tarefa 3 Construindo novos DataFrames

### Primeiro com a função concat fazemos a união dos 3 DataFrames df18, df19 e df20.

In [13]:
df_dados=pd.concat([df18,df19,df20],axis=0,	ignore_index=True)
df_dados

id data_inversa     dia_semana   horario  uf     br     km  \
0        99973.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
1        99976.0   2018-01-01  segunda-feira  00:40:00  SC  282.0    0,4   
2        99977.0   2018-01-01  segunda-feira  00:30:00  RJ  493.0      1   
3        99981.0   2018-01-01  segunda-feira  01:15:00  RS  386.0    134   
4        99982.0   2018-01-01  segunda-feira  00:20:00  RS  293.0  151,7   
...          ...          ...            ...       ...  ..    ...    ...   
200284  364615.0   2020-03-20    sexta-feira  09:00:00  RN  304.0  171,8   
200285  364810.0   2020-09-12         sábado  23:50:00  BA  116.0   11,5   
200286  364817.0   2020-10-30    sexta-feira  10:30:00  BA  116.0   14,3   
200287  364835.0   2020-09-20        domingo  23:00:00  BA  116.0      8   
200288  364903.0   2020-10-31         sábado  02:10:00  BA  116.0      3   

                   municipio                      causa_acidente  \
0                    RESENDE                   Condutor Dormindo   
1              FLORIANOPOLIS  Não guardar distância de segurança   
2                   ITABORAI              Ultrapassagem Indevida   
3                    SARANDI                  Ingestão de Álcool   
4                   CANDIOTA         Falta de Atenção à Condução   
...                      ...                                 ...   
200284               ANGICOS         Falta de Atenção à Condução   
200285  VITORIA DA CONQUISTA                  Ingestão de Álcool   
200286  VITORIA DA CONQUISTA         Falta de Atenção à Condução   
200287  VITORIA DA CONQUISTA             Velocidade Incompatível   
200288  VITORIA DA CONQUISTA                    Animais na Pista   

                       tipo_acidente  ... feridos_graves ilesos ignorados  \
0          Saída de leito carroçável  ...              4      1         0   
1                   Colisão traseira  ...              2      1         0   
2                    Colisão frontal  ...              0      3         1   
3                Colisão transversal  ...              0      2         0   
4          Saída de leito carroçável  ...              1      0         0   
...                              ...  ...            ...    ...       ...   
200284              Colisão traseira  ...              1      2         0   
200285               Colisão lateral  ...              0      2         0   
200286               Colisão lateral  ...              0      2         0   
200287  Queda de ocupante de veículo  ...              0      0         0   
200288       Atropelamento de Animal  ...              1      0         0   

       feridos veiculos      latitude     longitude  regional  delegacia  \
0            4        1     -22,46937     -44,44705     SR-RJ     DEL5/7   
1            2        2  -27,59971687  -48,57565694     SR-SC     DEL8/1   
2            1        3    -22,763901    -42,927532     SR-RJ     DEL5/4   
3            0        2    -27,953636    -52,916374     SR-RS    DEL9/14   
4            1        1    -31,395214  -53,78391174     SR-RS    DEL9/11   
...        ...      ...           ...           ...       ...        ...   
200284       2        4   -5,68711053    -36,419458     SR-RN      UOP03   
200285       1        2  -14,87308214  -40,87141514     SR-BA      UOP01   
200286       2        2  -14,89597653  -40,86098671     SR-BA      UOP01   
200287       1        1  -14,84732305  -40,88634968     SR-BA      UOP01   
200288       1        1    -14,824848    -40,861311     SR-BA      UOP01   

             uop  
0       UOP03/RJ  
1       UOP01/SC  
2       UOP02/RJ  
3       UOP01/RS  
4       UOP03/RS  
...          ...  
200284     DEL01  
200285     DEL08  
200286     DEL08  
200287     DEL08  
200288     DEL08  

[200289 rows x 30 columns]

# a) ranking de acidentes por estado para todos os anos combinados

In [14]:
ranking=df_dados['uf'].value_counts()
ranking.rank
df3=pd.DataFrame(ranking)
df3.rename(columns={'uf':'acidentes'}, inplace=True)
df3.index.names = ['Municipios']
df3.to_csv('Ranking.csv', sep='\t', encoding='utf-8')
df3

acidentes
Municipios           
MG              26160
SC              24145
PR              22850
RJ              13417
RS              13216
SP              12936
BA              10482
GO              10046
ES               8018
PE               8011
MT               6963
CE               4962
RO               4648
PB               4553
MS               4505
RN               4190
PI               3920
MA               3501
PA               3040
DF               2997
AL               1948
SE               1746
TO               1704
AC                789
RR                723
AP                456
AM                363

# b) acidentes por dia da semana para todos os anos combinados

In [15]:
acid=df_dados['dia_semana'].value_counts()
acidentes=pd.DataFrame(acid)
acidentes.rename(columns={'dia_semana':'acidentes'}, inplace=True)
acidentes.index.names = ['Dias da Semana']
acidentes.to_csv('Dias_da_Semana.csv', sep='\t', encoding='utf-8')
acidentes

acidentes
Dias da Semana           
sábado              33005
domingo             33001
sexta-feira         30777
segunda-feira       27075
quinta-feira        26229
quarta-feira        25353
terça-feira         24849

# Tarefa 4 Ocorrencia de acidentes

# a) qual a causa mais frequente e a mais rara de acidentes registrados nos três anos?

In [17]:
df_completo = pd.concat([df20,df19,df18])
causas_acidentes = df_completo['causa_acidente'].value_counts()
causas_acidentes = pd.DataFrame(causas_acidentes,columns=["causa_acidente"])
causas_acidentes = causas_acidentes.rename(columns={'causa_acidente': 'Quantidade de Acidentes'})
causas_acidentes['causa_acidente'] = causas_acidentes.index
causas_acidentes = causas_acidentes.set_index(['causa_acidente'])
causas_acidentes.head(1)

Quantidade de Acidentes
causa_acidente                                      
Falta de Atenção à Condução                    73231

### A causa mais frequente é falta de atenção á condução com 73231 casos.

### E as causas mais raras com somente 1 ocorrencia esta na tabela abaixo.

In [18]:
causas_acidentes.tail(12)

Quantidade de Acidentes
causa_acidente                                                             
Acumulo de areia ou detritos sobre o pavimento                            1
Pista esburacada                                                          1
Estacionar ou parar em local proibido                                     1
Fumaça                                                                    1
Pedestre cruzava a pista fora da faixa                                    1
Acostamento em desnível                                                   1
Condutor usando celular                                                   1
Frear bruscamente                                                         1
Ausência de sinalização                                                   1
Acumulo de óleo sobre o pavimento                                         1
Ingestão de álcool ou de substâncias psicoativa...                        1
Curva acentuada                                                           1

# b) qual a proporção de pessoas ilesas e de feridos graves por mês em cada um dos anos considerados? 

# c) mostre, reorganizando o conjunto de dados, se a pandemia conseguiu diminuir ou não a incidência de acidentes nas rodovias federais

In [19]:
df4=pd.DataFrame({"Acidentes2018":[len(df18)],
"Acidentes2019":[len(df19)],"Acidentes2020":[len(df20)]})
df4

Acidentes2018  Acidentes2019  Acidentes2020
0          69295          67446          63548

### Podemos ver que no ano de 2020 houve queda no numero de acidentes nas rodovias em comparação com os anos anteriores, com isso podemos deduzir que a pandemia conseguiu sim diminuir as incidencias de acidentes.